app/models/submission_template.rb

```diff
@@ -102,10 +102,13 @@ def input_plate_purposes
 
   private
 
+  # Retrieves the request types that are associated with this submission template,
+  # from the ids that are specified in the submission_parameters field.
   def request_types
     @request_types ||= RequestType.where(id: request_type_ids)
   end
 
+  # Returns the request types in the order that they are specified in the submission_parameters field.
```

Comment:
These are just unrelated comments I added locally a while ago while trying to understand this bit of code.